Framework/DataObjects/inc/MantidDataObjects/MDGridBox.tcc

```diff
@@ -1134,17 +1134,49 @@ TMDE(void MDGridBox)::integrateSphere(
     API::CoordTransform &radiusTransform, const coord_t radiusSquared,
     signal_t &signal, signal_t &errorSquared, const coord_t innerRadiusSquared,
     const bool useOnePercentBackgroundCorrection) const {
-  // We start by looking at the vertices at every corner of every box contained,
-  // to see which boxes are partially contained/fully contained.
+
+  // --------
+  // Overview
+  //   R: peak radius
+  //   r: inner radius (useful for defining peak background)
+  //   d: distance computed with radiusTransfor
+  // all distances are compared with their squared form
+  // ---------------------------------------------------------------------------
+  // We are checking all corners of each box to see if
+  //    ! all corners are within the integration volume (sphere/ellipsoid/donut)
+  //       -> we reach the base case, add box_signal and error to the container
+  //          (signal, errorSquared)
+  //    ! at least one corner is within the integration volume
+  //       -> recursively perform the same corner check for all its child boxes
+  //          util we reach the base case
+  //    ! no corner is within the integration volume
+  //       -> compute distance (d) between peak center and
+  //          [a] box center
+  //          [b] box corners
+  //          if d_min > R: box completely outside the integration volume, skip
+  //                        it and its children to save computing time
+  //          if d_max < r: box completely falls into the donut hole, skip it
+  //                        and its children to save computing time
+  //          others: recursively search the child boxes as the current box is
+  //                  partially overlapping with the integration volume, in
+  //                  otherwords, some of its child box is within the region
+  //                  (base case), and some of its child box is outside the
+  //                  integration volume.
 
   // One entry with the # of vertices in this box contained; start at 0.
-  auto verticesContained = new size_t[numBoxes];
-  memset(verticesContained, 0, numBoxes * sizeof(size_t));
+  std::vector<size_t> verticesContained(numBoxes, 0);
+
+  // Setup caches for distances dmax and dmin
+  auto distmaxs = new coord_t[numBoxes];
+  memset(distmaxs, 0, numBoxes * sizeof(coord_t));
+  auto distmins = new coord_t[numBoxes];
+  memset(distmins, 65535, numBoxes * sizeof(coord_t));
+  // std::vector<coord_t> distmins(numBoxes, radiusSquared + 1);
```

Comment:
This is very strange. The vector should act exactly as the array does - it feels like there is something about this code we don't fully understand. I don't want to hold up the bug fix but I think this should be investigated at some point and I think that would have to be done by stepping through with a debugger to see what operates differently in both scenarios.

Reply:
Could we remove the commented out code for now to avoid confusion for future developers?

Reply:
@KedoKudo `memset(distmins, 65535, numBoxes * sizeof(coord_t));` initializes the array as `-nan` and then every comparison with `distmins` will be `false`. I don't think this was what you were going for.

You can make using the vectors pass the unit tests If instead of

```cpp
std::vector<coord_t> distmins(numBoxes, radiusSquared + 1);
std::vector<coord_t> distmaxs(numBoxes, 0.0);
```
you do
```cpp
std::vector<coord_t> distmins(numBoxes, 0.0);
std::vector<coord_t> distmaxs(numBoxes, 0.0);
```
which suggest there is something wrong with the logic using `distmins`.

Reply:
Thanks @rosswhitfield. I didn't spot the bad initialization there - another reason to avoid C constructs!